master/internal/configpolicy/task_config_policy.go

```diff
@@ -160,6 +163,53 @@ func GetMergedConstraints(ctx context.Context, workspaceID int, workloadType str
 	return &constraints, nil
 }
 
+// MergeWithInvariantExperimentConfigs merges the config with workspace and global invariant
+// configs, where a global invariant config takes precedence over a workspace-level invariant
+// config.
+func MergeWithInvariantExperimentConfigs(ctx context.Context, workspaceID int,
+	config *expconf.ExperimentConfigV0,
+) error {
+	originalConfig := *config
+	var wkspOverride, globalOverride bool
+	wkspConfigPolicies, err := GetTaskConfigPolicies(ctx, &workspaceID, model.ExperimentType)
+	if err != nil {
+		return err
+	}
+	if wkspConfigPolicies.InvariantConfig != nil {
+		if err := json.Unmarshal([]byte(*wkspConfigPolicies.InvariantConfig), config); err != nil {
+			return fmt.Errorf("error unmarshaling workspace invariant config: %w", err)
+		}
+		wkspOverride = true
+	}
+
+	globalConfigPolicies, err := GetTaskConfigPolicies(ctx, nil, model.ExperimentType)
+	if err != nil {
+		return err
+	}
+	if globalConfigPolicies.InvariantConfig != nil {
+		if err = json.Unmarshal([]byte(*globalConfigPolicies.InvariantConfig), config); err != nil {
+			return fmt.Errorf(" error unmarshaling global invariant config: %w", err)
+		}
+	}
+
+	scope := ""
+	if wkspOverride {
+		if globalOverride {
+			scope += "workspace and global"
+		} else {
+			scope += "workspace"
+		}
+	} else if globalOverride {
+		scope += "global"
+	}
+
+	if !reflect.DeepEqual(originalConfig, *config) {
+		log.Warnf("some fields of submitted config were silently overrideen with %s config(s)",
```

Comment:
I think we want to put the phrase "config policies" here. Because users will view the config policies from a tab called "config policies" in the webUI, I think using that specific phrase will help them understand what happened and know where to look to find these config policies. Maybe try  something like `some fields were modified by admin %s  config policies`?

nit: typo in "overrideen"

Reply:
Gotcha! Makes sense to give a guiding warn. Changed this to `some fields were overridden by admin %s policies`